libevm/JitVM.cpp

```diff
@@ -248,47 +242,112 @@ int64_t evm_call(
 	return cost;
 }
 
+
+/// RAII wrapper for an evm instance.
+class EVM
+{
+	evm_instance* m_instance = nullptr;
+
+public:
+	EVM(evm_query_fn _queryFn, evm_update_fn _updateFn, evm_call_fn _callFn)
+	{
+		m_instance = evm_create(_queryFn, _updateFn, _callFn);
+	}
+
+	~EVM()
+	{
+		evm_destroy(m_instance);
+	}
+
+	EVM(EVM const&) = delete;
+	EVM& operator=(EVM) = delete;
+
+	struct Result : evm_result
+	{
+		Result(evm_result const& _other)
+		{
+			// FIXME: Looks like iheritance from C struct is not very effective.
+			//        Composition should be better?
+			gas_left = _other.gas_left;
+			output_data = _other.output_data;
+			output_size = _other.output_size;
+			internal_memory = _other.internal_memory;
+		}
+
+		~Result()
+		{
+			// TODO: Decide when the result has to be destroyed.
+			evm_destroy_result(*this);
```

Comment:
`this` will be different from `_other` passed into the constructor, but also see the comment above.